golem-worker-executor-base/src/virtual_export_compat/http_incoming_handler.rs

```diff
@@ -0,0 +1,125 @@
+// Copyright 2024-2025 Golem Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::convert::Infallible;
+
+use crate::error::GolemError;
+use bytes::Bytes;
+use golem_common::virtual_exports::http_incoming_handler::*;
+use golem_wasm_rpc::Value;
+use http::{HeaderName, HeaderValue};
+use http_body_util::combinators::BoxBody;
+use http_body_util::BodyExt;
+use wasmtime_wasi_http::bindings::http::types::ErrorCode;
+
+pub fn input_to_hyper_request(
```

Comment:
Not that it significantly matters, yet I feel like the module name should be  `*adapter`  (that converts to hyper request (required by wasihttpviewhandler::new_incoming_request) instead of a `handler`.

Reply:
I tried to express this using the compat part of the module naming.

I.e "this module is for compatibility with the http_incoming_handler virtual export"

Reply:
👍